pkgs/stdenv/generic/check-meta.nix

```diff
@@ -454,6 +464,15 @@ let
       hasOutput = out: builtins.elem out outputs;
     in
     {
+      # get the default value for the meta.repository field.
+      # the fetchFrom* fetchers set src.meta.homepage
+      # NOTE: this will fail if src fails to eval, in that case either set meta.repository manually to prevent this default from being evaluated, or just make sure src doesn't fail to eval.
```

Comment:
Have you tried `builtins.tryEval` to catch the `src` evaluation errors and just put nothing in it?

Reply:
builtins.tryEval only catches throw errors.  this note was orignally made when packages enconering abort-class errors in `src` was fairly common (at the time this was causing eval to fail), although both of these issues have been fixed since then.

Reply:
`abort`-class errors in evaluations are abnormal. Throwing in derivations is the only thing allowed to make evaluation fail.

Doing `src = { "x86_64-linux" = ...; }.${builtins.currentSystem}` is a mistake in nixpkgs, it requires at least:
`src = { ... }.${builtins.currentSystem} or (throw "Source not available on {...}")`

Thus, `tryEval` is the only thing you really really need in `checkMeta` for infaillible evaluations, all the other errors are *bugs* in nixpkgs and must be fixed as such because Hydra won't catch them neither.